airbyte_cdk/sources/declarative/declarative_component_schema.yaml

```diff
@@ -2974,6 +2976,52 @@ definitions:
       - type
       - retriever
       - components_mapping
+  StreamConfig:
```

Comment:
The naming is mildly confusing to me. It does not seem to be a config per se but more like something to figure out where in the source config we can find the stream config. Is there a reason why we created a component that only has one field? I'm wondering if we could just move the `configs_pointer` under `ConfigComponentsResolver`

Reply:
We can move the `configs_pointer` under `ConfigComponentsResolver`, but if any transformations or other specific details are needed for the config, we would have to add them at the `ConfigComponentsResolver` level, which isn't ideal. Having a separate entity, such as a `StreamConfig`, would make it easier to extend if needed.